lib/active_record_doctor/detectors/incorrect_dependent_option.rb

```diff
@@ -60,19 +63,29 @@ def detect
 
             deletable_models, destroyable_models = associated_models.partition { |klass| deletable?(klass) }
 
-            if callback_action(association) == :invoke && destroyable_models.empty? && deletable_models.present?
-              suggestion =
-                case association.macro
-                when :has_many then :suggest_delete_all
-                when :has_one, :belongs_to then :suggest_delete
-                else raise("unsupported association type #{association.macro}")
-                end
-
-              problem!(model: model.name, association: association.name, problem: suggestion,
-                       associated_models: deletable_models.map(&:name))
-            elsif callback_action(association) == :skip && destroyable_models.present?
-              problem!(model: model.name, association: association.name, problem: :suggest_destroy,
-                       associated_models: destroyable_models.map(&:name))
+            case association.options[:dependent]
+            when :destroy_async
+              foreign_key = foreign_key(association.klass.table_name, model.table_name)
+              if foreign_key
```

Comment:
It seems `foreign_key` can be inlined, however rewording the message may force use to keep that variable here. Leaving this up to you.